macOS nightly fails to build with new libmagic from vcpkg

I synced my fork today, and the macOS nightly [failed](https://github.com/jdblischak/TileDB-MariaDB/actions/runs/8251274256/job/22570743354#step:6:86) (even after a restart). It fails to install libmagic with vcpkg, which was just committed upstream about 8 hours ago in https://github.com/TileDB-Inc/TileDB/commit/7753f3360b42a837f2e18275a5b23691aa86157d (https://github.com/TileDB-Inc/TileDB/pull/4673).

I assume the fix involves installing/updating autoconf/autotools with homebrew

cc @teo-tsirpanis 

```
Installing 3/28 libmagic:x64-osx...
Building libmagic:x64-osx...
-- [OVERLAY] Loading triplet configuration from: /Users/runner/work/TileDB-MariaDB/TileDB-MariaDB/libtiledb/ports/triplets/x64-osx.cmake
-- Installing port from location: /Users/runner/work/TileDB-MariaDB/TileDB-MariaDB/libtiledb/ports/libmagic
-- Downloading https://github.com/file/file/archive/FILE5_45.tar.gz -> file-file-FILE5_45.tar.gz...
-- Extracting source /Users/runner/work/TileDB-MariaDB/TileDB-MariaDB/libtiledb/build/_deps/vcpkg-src/downloads/file-file-FILE5_45.tar.gz
-- Using source at /Users/runner/work/TileDB-MariaDB/TileDB-MariaDB/libtiledb/build/_deps/vcpkg-src/buildtrees/libmagic/src/FILE5_45-8b12c22[92](https://github.com/jdblischak/TileDB-MariaDB/actions/runs/8251274256/job/22570743354#step:6:93)1.clean
-- Getting CMake variables for x64-osx-dbg
-- Getting CMake variables for x64-osx-rel
-- Generating configure for x64-osx
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /usr/local/bin/autoreconf -vfi
    Working Directory: /Users/runner/work/TileDB-MariaDB/TileDB-MariaDB/libtiledb/build/_deps/vcpkg-src/buildtrees/libmagic/src/FILE5_45-8b12c22921.clean/
    Error code: 2
    See logs for more information:
      /Users/runner/work/TileDB-MariaDB/TileDB-MariaDB/libtiledb/build/_deps/vcpkg-src/buildtrees/libmagic/autoconf-x64-osx-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:629 (vcpkg_execute_required_process)
  /Users/runner/work/TileDB-MariaDB/TileDB-MariaDB/libtiledb/ports/libmagic/portfile.cmake:52 (vcpkg_configure_make)
  scripts/ports.cmake:170 (include)


error: building libmagic:x64-osx failed with: BUILD_FAILED
Elapsed time to handle libmagic:x64-osx: 4.5 s
```

--- a/.github/workflows/nightly.yaml
+++ b/.github/workflows/nightly.yaml
@@ -5,6 +5,10 @@ on:
     paths:
       - '.github/workflows/nightly.yaml'
       - 'scripts/ci/**'
+  pull_request:
+    paths:
+      - '.github/workflows/nightly.yaml'
+      - 'scripts/ci/**'
   schedule:
     - cron: "0 2 * * *" # Every night at 2 AM UTC (9 PM EST; 10 PM EDT)
   workflow_dispatch:

--- a/scripts/ci/setup-macos.sh
+++ b/scripts/ci/setup-macos.sh
@@ -5,6 +5,7 @@ export HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK=1
 
 # Install dependencies on macOS
 brew install \
+  automake \
   bison@2.7 \
   boost \
   cmake \

